### PR TITLE
Move rcharset from core to rprint

### PR DESCRIPTION
**Pull request by gogo2464**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

fix the first point of this comment: https://github.com/radareorg/radare2/pull/18028

### CODE REVIEW

File: libr/core/core.c [hunk 1]
```diff
@@ -2713,8 +2713,8 @@ R_API bool r_core_init(RCore *core) {
 	core->printidx = 0;
 	core->lastcmd = NULL;
 	core->cmdlog = NULL;
-	core->charset = r_charset_new ();
-	core->charset->db = sdb_ns (core->sdb, "charset", 1);
+	core->print->charset = r_charset_new ();
```
Comment (trufae) [inline libr/core/core.c:2716]: this should be moved into r_print_new()

File: libr/core/core.c [hunk 2]
```diff
@@ -2902,7 +2902,7 @@ R_API void r_core_fini(RCore *c) {
 	if (!c) {
 		return;
 	}
-	r_charset_free (c->charset); // TODO. maybe move into RPrint?
+	r_charset_free (c->print->charset);
```
Comment (trufae) [inline libr/core/core.c:2905]: move this into r_print_free

File: libr/core/core.c [hunk 3]
```diff
@@ -2713,8 +2713,8 @@ R_API bool r_core_init(RCore *core) {
 	core->printidx = 0;
 	core->lastcmd = NULL;
 	core->cmdlog = NULL;
-	core->charset = r_charset_new ();
-	core->charset->db = sdb_ns (core->sdb, "charset", 1);
+	core->print->charset = r_charset_new ();
+	core->print->charset->db = sdb_ns (core->sdb, "charset", 1);
```
Comment (trufae) [inline libr/core/core.c:2717]: as long as core is needed here to set the NS, i think its better to:

* in r_print_new() call print->charset->db = sdb_new0();
* in core_init() after calling r_print_new() use `sdb_ns_set (core->sdb, "charset", core->print->charset->db);`

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/core/cmd_print.c [hunk 4]
```diff
@@ -5833,6 +5832,24 @@ l = use_blocksize;
 				}
 			}
 			break;
+		case 'e': // "pse"
+			// should be done in `ps` when cfg.charset is set
+			if (len > 0) {
+				size_t out_len = len * 10;
+				ut8 *out = calloc (len, 10);
+				if (out) {
+					ut8 *data = calloc (1, len);
```
[the remark this answers is not in the document]
Comment (gogo2464) [reply libr/core/cmd_print.c:5849]: Strange! I did not make this change. I think I have either rebased or you have modified it when you move from `pse` to `e cfg.charset`. I will get a look at.

**Review by trufae — APPROVED** [review]

**Pull request merged**